public/app/features/apiserver/client.ts

```diff
@@ -19,47 +20,52 @@ import {
   K8sAPIGroupList,
   AnnoKeySavedFromUI,
   ResourceEvent,
+  GroupVersionResource,
 } from './types';
 
-export interface GroupVersionResource {
-  group: string;
-  version: string;
-  resource: string;
-}
-
 export class ScopedResourceClient<T = object, S = object, K = string> implements ResourceClient<T, S, K> {
   readonly url: string;
+  readonly gvr: GroupVersionResource;
 
   constructor(gvr: GroupVersionResource, namespaced = true) {
     const ns = namespaced ? `namespaces/${getAPINamespace()}/` : '';
-
+    this.gvr = gvr;
     this.url = `/apis/${gvr.group}/${gvr.version}/${ns}${gvr.resource}`;
   }
 
   public async get(name: string): Promise<Resource<T, S, K>> {
     return getBackendSrv().get<Resource<T, S, K>>(`${this.url}/${name}`);
   }
 
-  public watch(
-    params?: WatchOptions,
-    config?: Pick<BackendSrvRequest, 'data' | 'method'>
-  ): Observable<ResourceEvent<T, S, K>> {
+  public watch(params?: WatchOptions): Observable<ResourceEvent<T, S, K>> {
+    if (true) {
```

Comment:
@Clarity-89  --- for now I am replacing everything, longer term, we should only use websocket if not http2 (but there is no good way to know!)